### PR TITLE
h-include when

### PR DESCRIPTION
**Pull request by nicolasdelfino**
When offers a way of using a predicate for inclusion. It also features an optional alt attribute that functions as the source of inclusion given that the predicate fails.

<h-include src="logged-in.html" when="org.project.predicateFunction" alt="log-in.html"></h-include>

The method specified in the when attribute may be namespaced and needs to return true for the inclusion of the url specified in the src attribute to occur.

### CODE REVIEW

**Comment by gustafnk** [comment]
Thank you for your PR! :)

(It could be good to know for the rest of the internet that @nicolasdelfino and I have chatted quite extensively about this in other channels)

I have though a bit on the naming of the `alt` attribute and I thought that we (in the future) should implement non-when error handling using the `alt` attribute as well. @nicolasdelfino , it would be nice to get your thoughts on this (before merging).

---
In non-when-scenarios:
 - if AJAX from `src` attribute results in 4xx/5xx response => use `alt` attribute for AJAX

In when-scenarios:
- if `when` attribute is falsy OR if AJAX from `src` attribute results in 4xx/5xx response => use `alt` attribute for AJAX
---

Or, in one rule:
- if ('when' attribute AND `when` attribute is falsy) OR if AJAX from `src` attribute results in 4xx/5xx response => use `alt` attribute for AJAX

What do you think @nicolasdelfino ? :)

**Comment by nicolasdelfino** [comment]
I like it @gustafnk - it's a way of extending the when predicate to not only live in userland but also in h-include itself. 

The rule could basically be:
* if ajax results in 4/5... and alt is present, use alt. 
regardless if when is present or not.

Do you want me to update this pr or should we do another one in the future?

**Comment by gustafnk** [comment]
Let’s take that in another PR :)

I just want to marinate all this a little bit more though and have some time to look at the tests

File: lib/h-include.js [hunk 1]
```diff
@@ -139,6 +145,33 @@ window.HInclude.HIncludeElement = window.HIncludeElement = (function() {
 
   var proto = Object.create(HTMLElement.prototype);
 
+  function getPredicate(identifier, context) {
+    var namespaces = identifier.split(".");
```
Comment (gustafnk) [inline lib/h-include.js]: Please use single quotes

**Pull request merged**